src/shogun/kernel/CombinedKernel.cpp

```diff
@@ -169,15 +171,100 @@ bool CCombinedKernel::init(CFeatures* l, CFeatures* r)
 		return false;
 	}
 
-	if ( ((CCombinedFeatures*) l)->get_num_feature_obj()<=0 ||
-		((CCombinedFeatures*) l)->get_num_feature_obj() != ((CCombinedFeatures*) r)->get_num_feature_obj() )
-		SG_ERROR("CombinedKernel: Number of features/kernels does not match - bailing out\n")
+	if (((CCombinedFeatures*)l)->get_num_feature_obj() <= 0 ||
```

Comment:
you can dynamic cast these into a var and then use them... one would of course suppose that the compiler would do this as well under the hood... but it might be easier to read the code...?